src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -464,6 +464,24 @@ QgsOgrProvider::QgsOgrProvider( QString const &uri, const ProviderOptions &optio
   }
   setNativeTypes( nativeTypes );
 
+  if ( mOgrOrigLayer )
+  {
+    bool is3D = true;
+    gdal::ogr_feature_unique_ptr f;
+
+    mOgrOrigLayer->ResetReading();
```

Comment:
Why are we sniffing a feature here instead of just checking the wkb type of the layer?

Reply:
> Why are we sniffing a feature here instead of just checking the wkb type of the layer?

That was my initial reaction, but there are a number of data sources where the typing is loose. That said the strategy could be :
1. look at the wkb type of the layer: if it is 3D, then note it and stop
2. otherwise sniff a feature

Reply:
fixed!